pkg/loadbalancer/experimental/reflector.go

```diff
@@ -286,11 +293,11 @@ func runServiceEndpointsReflector(ctx context.Context, health cell.Health, p ref
 			p.waitTime(),
 			func(buf buffer, ev resource.Event[runtime.Object]) buffer {
 				if buf == nil {
-					buf = map[bufferKey]resource.Event[runtime.Object]{}
+					buf = container.NewInsertOrderedMap[bufferKey, resource.Event[runtime.Object]](reflectorInitialBufferSize)
```

Comment:
Bit confused, why isn't the goal to process both Pods and Service events in their relative order (i.e. from what I can tell - this is only processing Pods in the order we saw them in).

Should we need to serialize Pod/Svc updates into the same process loop?

Reply:
We process services and endpoint slices in shared buffer to reduce the amount of times we commit services without endpoints or other way around. Pods are for HostPort and there we create a service, frontend and backends out of a single pod, so there's no data dependency. Simpler to not combine that with the service & eps processing.

Reply:
Ah yeah I misunderstood the code.  What I was getting was correct (swap out Pods for Endpoints). I see below in the processBuffer fn that the buffer is processed in the relative order I would expect.